docs/docs/ref-01-top-level-api.md

```diff
@@ -161,6 +161,10 @@ If the optional callback is provided, it will be executed after the component is
 > `ReactDOM.render()` does not modify the container node (only modifies the children of the container). In
 > the future, it may be possible to insert a component to an existing DOM node without overwriting
 > the existing children.
+>
+> `ReactDOM.render()` currently returns a reference to the root ReactComponent instance.  However, using this return value is legacy
+> and should be avoided.  If you need a reference to the root ReactComponent instance, the preferred solution is to attach a
+> [callback ref](/react/docs/more-about-refs.html#the-ref-callback-attribute) to the root element.
```

Comment:
For the purposes of getting a reference to the root node, I think they're both semantically equivilant.  The optional callback was introduced before callback refs were a thing.  For consistency, I think we would want to teach refs as the standard way of getting a reference to an instance.

However, callback refs and the optional callback are not identical with regards to their other semantics.  The optional callback will be fired when the render completes, whereas the callback ref will only be fired when a component mounts/unmounts.  So their behavior is the same for initial mount, but different for updates.  Realistically, now that we have callback refs, the optional callback probably isn't very useful, but there are still some potential use cases if you wanted to know when a particular update/render has fully completed.